src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesReconciliationService.kt

```diff
@@ -119,32 +119,34 @@ class CaseNotesReconciliationService(
   suspend fun checkMatchOrThrowException(offenderNo: String): MismatchCaseNote? {
     val mappings = caseNotesMappingApiService.getByPrisoner(offenderNo).mappings
 
-    val dpsDistinctIds = mappings.map { it.dpsCaseNoteId }.distinct()
+    val mappingsDpsDistinctIds = mappings.map { it.dpsCaseNoteId }.distinct()
 
     val dpsCaseNotes = caseNotesDpsApiService.getCaseNotesForPrisoner(offenderNo)
       .associate { it.caseNoteId to it.transformFromDps() }
 
     val nomisCaseNotes = caseNotesNomisApiService.getCaseNotesForPrisoner(offenderNo).caseNotes
       .associate { it.caseNoteId to it.transformFromNomis() }
 
-    val message = "mappings.size = ${mappings.size}, dpsDistinctIds.size = ${dpsDistinctIds.size}, nomisCaseNotes.size = ${nomisCaseNotes.size}, dpsCaseNotes.size = ${dpsCaseNotes.size}"
+    val message = "mappings.size = ${mappings.size}, mappingsDpsDistinctIds.size = ${mappingsDpsDistinctIds.size}, nomisCaseNotes.size = ${nomisCaseNotes.size}, dpsCaseNotes.size = ${dpsCaseNotes.size}"
     val mismatchCaseNote = MismatchCaseNote(offenderNo, emptySet(), emptySet())
 
     if (mappings.size != nomisCaseNotes.size) {
       log.info("prisoner $offenderNo : $message")
+      val differences = mappings.map { it.nomisCaseNoteId }.toSet().xor(nomisCaseNotes.map { it.key }.toSet())
```

Comment:
find the case note ids that are only in each set and not in the intersection of them - which will tell us which ones are different between the two